malcolm/core/process.py

```diff
@@ -91,3 +108,52 @@ def spawn(self, function, *args, **kwargs):
         self._other_spawned.append(spawned)
         return spawned
 
+    def _handle_block_notify(self, request):
+        for subscription in self._subscriptions:
+            # find stuff that's changed that is relevant to this subscriber
+            endpoint = subscription.endpoint
+            changes = []
+            if len(endpoint) == 0:
+                changes = self._last_changes
+            else:
+                for change in self._last_changes:
+                    i = 0
+                    for (change_path, sub_path) in zip(change[0], endpoint):
+                        if change_path != sub_path:
+                            break
+                        i += 1
+                    else:
+                        # change has matching path, so keep it
+                        # but strip off the end point path
+                        filtered_change = [change[0][i:], change[1]]
+                        changes.append(filtered_change)
+            if subscription.delta:
+                subscription.response_queue.put(changes)
+            elif len(changes) > 0:
+                # return the structure of everything below the endpoint
+                update = self._block_state_cache
+                for p in endpoint:
+                    update = update[p]
+                subscription.response_queue.put(update)
+
+    def _handle_block_changed(self, request):
+        for path, value in request.changes:
+            # update changes
+            for e in self._last_changes:
+                if e[0] == path:
+                    e[1] = value
+                    break
+            else:
+                self._last_changes.append([path, value])
+
+            # update cached dict
```

Comment:
I think this code belongs in block_notify, so that anyone doing a get gets the last consistent serialized structure